### ISSUE TITLE
Grafana 7.2.1: Migration Error on restarting/scaling Grafana Pods: "Error 1054: Unknown column 'account_id' in 'field list'"

### ISSUE THREAD

**Issue by o-otte** [grafana/grafana#28218]
**What happened**:
We want to use Grafana 7.2.1 in our AKS Cluster to provide scalability. We are using an Azure Database for MySQL Server as the DB Backend and when scaling the Deployment, the new pods run the DB Migration process and fail with the following log:
```
t=2020-10-13T11:40:15+0000 lvl=info msg="Starting Grafana" logger=server version=7.2.1 commit=72a6c64532 branch=HEAD compiled=2020-10-08T09:00:32+0000
t=2020-10-13T11:40:15+0000 lvl=info msg="Config loaded from" logger=settings file=/usr/share/grafana/conf/defaults.ini
t=2020-10-13T11:40:15+0000 lvl=info msg="Config loaded from" logger=settings file=/etc/grafana/grafana.ini
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.data=/var/lib/grafana"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.logs=/var/log/grafana"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.plugins=/var/lib/grafana/plugins"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.provisioning=/opt"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from command line" logger=settings arg="default.log.mode=console"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_PATHS_DATA=/var/lib/grafana"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_PATHS_LOGS=/var/log/grafana"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_PATHS_PLUGINS=/var/lib/grafana/plugins"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_PATHS_PROVISIONING=/opt"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_SERVER_ROOT_URL=https://our.sub.domain.tld/mski"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_SERVER_SERVE_FROM_SUB_PATH=true"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_DATABASE_TYPE=mysql"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_DATABASE_HOST=dbserver.mysql.database.azure.com"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_DATABASE_NAME=databasename?allowCleartextPasswords=true&collation=utf8mb4_unicode_ci&tls=custom#"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_DATABASE_USER=databaseuser@databaseserver"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_DATABASE_PASSWORD=*********"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_DATABASE_SSL_MODE=true"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_DATABASE_CA_CERT_PATH=/etc/ssl/cert.pem"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_DATABASE_SERVER_CERT_NAME=databaseserver.mysql.database.azure.com"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_SECURITY_DISABLE_INITIAL_ADMIN_CREATION=false"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_SECURITY_ADMIN_USER=admin"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_SECURITY_ADMIN_PASSWORD=*********"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_SECURITY_SECRET_KEY=*********"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_AUTH_AZUREAD_NAME=Azure AD"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_AUTH_AZUREAD_ENABLED=true"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_AUTH_AZUREAD_ALLOW_SIGN_UP=true"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_AUTH_AZUREAD_CLIENT_ID=***********Deleted-for-privacy********"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_AUTH_AZUREAD_CLIENT_SECRET=*********"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_AUTH_AZUREAD_SCOPES=openid email profile"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_AUTH_AZUREAD_AUTH_URL=https://login.microsoftonline.com/********ID Removed*******/oauth2/v2.0/authorize"
t=2020-10-13T11:40:15+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_AUTH_AZUREAD_TOKEN_URL=https://login.microsoftonline.com/********ID Removed*******/oauth2/v2.0/token"
t=2020-10-13T11:40:15+0000 lvl=info msg="Path Home" logger=settings path=/usr/share/grafana
t=2020-10-13T11:40:15+0000 lvl=info msg="Path Data" logger=settings path=/var/lib/grafana
t=2020-10-13T11:40:15+0000 lvl=info msg="Path Logs" logger=settings path=/var/log/grafana
t=2020-10-13T11:40:15+0000 lvl=info msg="Path Plugins" logger=settings path=/var/lib/grafana/plugins
t=2020-10-13T11:40:15+0000 lvl=info msg="Path Provisioning" logger=settings path=/opt
t=2020-10-13T11:40:15+0000 lvl=info msg="App mode production" logger=settings
t=2020-10-13T11:40:15+0000 lvl=info msg="Connecting to DB" logger=sqlstore dbtype=mysql
t=2020-10-13T11:40:15+0000 lvl=info msg="Starting DB migrations" logger=migrator
t=2020-10-13T11:40:16+0000 lvl=info msg="Executing migration" logger=migrator id="create migration_log table"
t=2020-10-13T11:40:16+0000 lvl=info msg="Executing migration" logger=migrator id="create user table"
t=2020-10-13T11:40:16+0000 lvl=info msg="Executing migration" logger=migrator id="add unique index user.login"
t=2020-10-13T11:40:16+0000 lvl=warn msg="Skipping migration: Already executed, but not recorded in migration log" logger=migrator id="add unique index user.login"
t=2020-10-13T11:40:16+0000 lvl=info msg="Executing migration" logger=migrator id="add unique index user.email"
t=2020-10-13T11:40:16+0000 lvl=warn msg="Skipping migration: Already executed, but not recorded in migration log" logger=migrator id="add unique index user.email"
t=2020-10-13T11:40:16+0000 lvl=info msg="Executing migration" logger=migrator id="drop index UQE_user_login - v1"
t=2020-10-13T11:40:16+0000 lvl=info msg="Executing migration" logger=migrator id="drop index UQE_user_email - v1"
t=2020-10-13T11:40:17+0000 lvl=info msg="Executing migration" logger=migrator id="Rename table user to user_v1 - v1"
t=2020-10-13T11:40:17+0000 lvl=info msg="Executing migration" logger=migrator id="create user table v2"
t=2020-10-13T11:40:17+0000 lvl=info msg="Executing migration" logger=migrator id="create index UQE_user_login - v2"
t=2020-10-13T11:40:18+0000 lvl=info msg="Executing migration" logger=migrator id="create index UQE_user_email - v2"
t=2020-10-13T11:40:18+0000 lvl=info msg="Executing migration" logger=migrator id="copy data_source v1 to v2"
t=2020-10-13T11:40:19+0000 lvl=eror msg="Executing migration failed" logger=migrator id="copy data_source v1 to v2" error="Error 1054: Unknown column 'account_id' in 'field list'"
t=2020-10-13T11:40:19+0000 lvl=eror msg="Exec failed" logger=migrator error="Error 1054: Unknown column 'account_id' in 'field list'" sql="INSERT INTO `user` (`id`\n, `version`\n, `org_id`\n, `updated`\n, `rands`\n, `company`\n, `is_admin`\n, `login`\n, `email`\n, `name`\n, `password`\n, `salt`\n, `created`) SELECT `id`\n, `version`\n, `account_id`\n, `updated`\n, `rands`\n, `company`\n, `is_admin`\n, `login`\n, `email`\n, `name`\n, `password`\n, `salt`\n, `created` FROM `user_v1`"
t=2020-10-13T11:40:19+0000 lvl=eror msg="Server shutdown" logger=server reason="Service init failed: migration failed: migration failed: Failed to roll back transaction due to error: %!s(<nil>): Error 1054: Unknown column 'account_id' in 'field list'"
```
The first pod that gets the fresh and empty database runs the migrations smoothly and without problems. After the first Grafana Pod is ready and the next Pod is schedueled by scaling, the second Pod has the log posted above. The first Pod at this stage logs the following message:
```
t=2020-10-13T11:44:21+0000 lvl=eror msg="Failed to get system stats" logger=metrics error="Error 1054: Unknown column 'last_seen_at' in 'where clause'"
```
After that the Admin User is not capable to log in.

**What you expected to happen**:
Multiple Grafana Pod replicas that uses the same MySQL Database and that the Deployment can be scaled, or redeployed safely.

**How to reproduce it (as minimally and precisely as possible)**:
Running multiple (one after another) Grafana Pods in Kubernetes, using a shared MySQL Database PaaS from Azure as the DB Backend. DB User Authentication is done by using Azure managed identities.

**Anything else we need to know?**:
Data-Sources and Dashboards are provisioned by an initContainer.

This is the Migration Log after the first Grafana Pod is running:

**Environment**:
- Grafana version: 7.2.1
- Kubernetes version: 1.16.15 (AKS Azure Managed Kubernetes Cluster)
- MySQL Version: 5.7 authentication based on AAD Users and managed Identities.
- Data source type & version:
- OS Grafana is installed on:  Linux grafana-7bc54dfb67-nhb69 4.15.0-1096-azure #106~16.04.1-Ubuntu  x86_64 Linux
- User OS & Browser:
- Grafana plugins:
- Others:
[migrationlog.txt](https://github.com/grafana/grafana/files/5371483/migrationlog.txt)

**Comment by marefr** [comment]
No other similar reports as far as I know. Based on the log It looks like you've partially have run the migration, but failed to record it in the migration_log table. Please check the content of migration_log table - every successful migration should have success column set to 1.

Could this be related to your running mysql in HA mode with multiple instance, replication and/or multi-master setup (#13497)?

**Comment by o-otte** [comment]
We only used a single MySQL server without any replication, multi-master, or multiple instances so I think it has nothing to do with the mentioned issue.

After going through every configuration detail, I found out that it seems to be a problem when using MySQL users that are identified by Azure managed identities, when using the AAD Integration for the Azure Database for MySQL PaaS. See [Use Azure Active Directory for authentication with MySQL](https://docs.microsoft.com/en-us/azure/mysql/howto-configure-sign-in-azure-ad-authentication) and [Connect with Managed Identity to Azure Database for MySQL](https://docs.microsoft.com/en-us/azure/mysql/howto-connect-with-managed-identity).

I didn't have time to investigate this further, yet. But it worked, when using normal MySQL Users on the same database and server.

For the migration log, it showed that the first migration of the first grafana pod worked without problems, but for the migration of the second pod, that it crashed in the way I mentioned in the initial post.
